examples/6field-simple/elm_6f.cxx

```diff
@@ -12,1677 +12,1682 @@
-  viscos_perp = options["viscos_perp"].withDefault(-1.0); // Perpendicular viscosity
-  hyperviscos = options["hyperviscos"].withDefault(-1.0); // Radial hyperviscosity
-
-  // Parallel temperature diffusion
-  diffusion_par = options["diffusion_par"].withDefault(-1.0);
-  // M: 4th Parallel density diffusion
-  diffusion_n4 = options["diffusion_n4"].withDefault(-1.0);
-  // M: 4th Parallel ion temperature diffusion
-  diffusion_ti4 = options["diffusion_ti4"].withDefault(-1.0);
-  // M: 4th Parallel electron temperature diffusion
-  diffusion_te4 = options["diffusion_te4"].withDefault(-1.0);
-  // M: 4th Parallel ion parallel velocity diffusion
-  diffusion_v4 = options["diffusion_v4"].withDefault(-1.0);
-  // xqx: parallel hyper-viscous diffusion for vorticity
-  diffusion_u4 = options["diffusion_u4"].withDefault(-1.0);
-
-  // heating factor in pressure
-  // heating power in pressure
-  heating_P = options["heating_P"].withDefault(-1.0);
-  // the percentage of radial grid points for heating profile radial
-  // width in pressure
-  hp_width = options["hp_width"].withDefault(0.1);
-  // the percentage of radial grid points for heating profile radial
-  // domain in pressure
-  hp_length = options["hp_length"].withDefault(0.04);
-
-  // sink factor in pressure
-  // sink in pressure
-  sink_vp = options["sink_vp"].withDefault(-1.0);
-  // the percentage of radial grid points for sink profile radial
-  // width in pressure
-  sp_width = options["sp_width"].withDefault(0.05);
-  // the percentage of radial grid points for sink profile radial
-  // domain in pressure
-  sp_length = options["sp_length"].withDefault(0.04);
-
-  // left edge sink factor in vorticity
-  // left edge sink in vorticity
-  sink_Ul = options["sink_Ul"].withDefault(-1.0);
-  // the percentage of left edge radial grid points for sink profile
-  // radial width in vorticity
-  su_widthl = options["su_widthl"].withDefault(0.06);
-  // the percentage of left edge radial grid points for sink profile
-  // radial domain in vorticity
-  su_lengthl = options["su_lengthl"].withDefault(0.15);
-
-  // right edge sink factor in vorticity
-  // right edge sink in vorticity
-  sink_Ur = options["sink_Ur"].withDefault(-1.0);
-  // the percentage of right edge radial grid points for sink profile
-  // radial width in vorticity
-  su_widthr = options["su_widthr"].withDefault(0.06);
-  // the percentage of right edge radial grid points for sink profile
-  // radial domain in vorticity
-  su_lengthr = options["su_lengthr"].withDefault(0.15);
-
-  // Compressional terms
-  phi_curv = options["phi_curv"].withDefault(true);
-  g = options["gamma"].withDefault(5.0 / 3.0);
-
-  if (!include_curvature)
-    b0xcv = 0.0;
-
-  if (!include_jpar0)
-    J0 = 0.0;
-
-  if (noshear) {
-    if (include_curvature)
-      b0xcv.z += I * b0xcv.x;
-    I = 0.0;
-  }
+protected:
+  int init(bool restarting) override {
+    bool noshear;
```

Comment:
warning: variable `noshear` is not initialized [cppcoreguidelines-init-variables]

```cpp
bool noshear;
     ^
             = 0
```